src/tm-threads.c

```diff
@@ -1713,16 +1770,32 @@ TmEcode TmThreadSpawn(ThreadVars *tv)
         return TM_ECODE_FAILED;
     }
 
-    /* Initialize and set thread detached attribute */
-    pthread_attr_init(&attr);
+#ifdef HAVE_DPDK
+    /* only DPDK worker threads are launched using rte_eal_remote_launch */
+    if (tv->lcore_id != LCORE_ID_ANY) {
+        struct dpdk_callback cb = { .func = tv->tm_func, .tv = tv };
 
-    pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
+        int rc = rte_eal_remote_launch(&TmThreadSpawnDpdkThread, (void *)&cb, tv->lcore_id);
```

Comment:
Suricata provides API calls which launches on the worker therads as defined in YAML file. please rework this logic

Reply:
I think I am a bit lost here. Could you please be more specific? I am not sure what API calls or what parts of YAML file you mean. 
`TmThreadSpawnDpdkThread` is a wrapper to run tv->tm_func because `rte_eal_remote_launch` accepts a different function pointer compared to `pthread_create`. It does not do anything more than that. 
CPU affinity is reassigned later in the thread initialization process if defined differently than `tv->lcore_id`.
Thanks for explanation.